clients/vscode/src/lsp/CodeLensMiddleware.ts

```diff
@@ -39,13 +39,23 @@ const decorationTypePending = window.createTextEditorDecorationType({
   isWholeLine: true,
   rangeBehavior: DecorationRangeBehavior.ClosedClosed,
 });
-const decorationTypeInserted = window.createTextEditorDecorationType({
+const decorationTypeTextInserted = window.createTextEditorDecorationType({
   backgroundColor: new ThemeColor("diffEditor.insertedTextBackground"),
+  isWholeLine: false,
+  rangeBehavior: DecorationRangeBehavior.ClosedOpen,
+});
+const decorationTypeTextDeleted = window.createTextEditorDecorationType({
+  backgroundColor: new ThemeColor("diffEditor.removedTextBackground"),
+  isWholeLine: false,
+  rangeBehavior: DecorationRangeBehavior.ClosedOpen,
+});
+const decorationTypeLineInserted = window.createTextEditorDecorationType({
+  backgroundColor: new ThemeColor("diffEditor.insertedLineBackground"),
   isWholeLine: true,
   rangeBehavior: DecorationRangeBehavior.ClosedClosed,
 });
 const decorationTypeDeleted = window.createTextEditorDecorationType({
```

Comment:
Also rename this to keep consistent.
```suggestion
const decorationTypeLineDeleted = window.createTextEditorDecorationType({
```